src/routes/subscriptions.js

```diff
@@ -1,128 +1,31 @@
-    { fullUrl: planDefinitionFullUrl, ...planDefinition },
-    r => r.fullUrl === planDefinitionFullUrl
-  );
-  debug(`Fetched ${server.endpoint}/PlanDefinition/${planDefinition.id}`);
-
-  const token = await getAccessToken(server.endpoint);
-  const headers = { Authorization: `Bearer ${token}` };
-  axios.get(`${server.endpoint}/Endpoint/${endpointId}`, { headers: headers }).then(response => {
-    if (response.data) {
-      const endpoint = response.data;
-      const endpointFullUrl = `${server.endpoint}/Endpoint/${endpoint.id}`;
-      db.upsert(
-        ENDPOINTS,
-        { fullUrl: endpointFullUrl, ...endpoint },
-        r => r.fullUrl === endpointFullUrl
-      );
-      debug(`Fetched ${endpointFullUrl}`);
-    }
-  });
-
-  const subscriptions = subscriptionsFromPlanDef(planDefinition, server.endpoint);
-  postSubscriptionsToEHR(subscriptions);
-
-  res.sendStatus(StatusCodes.OK);
-});
+router.put('/ka/:id/:resource/:resourceId', notificationService.knowledgeArtifact);
 
 /**
  * Notification from EHR of changed data to start workflow process. The
  * id param is the PlanDefinition id which defines the workflow
  */
-router.post('/:fullUrl', (req, res) => {
-  const fullUrl = req.params.fullUrl;
-  const planDef = getPlanDef(fullUrl);
-  if (planDef) {
-    res.sendStatus(StatusCodes.OK);
-    startReportingWorkflow(planDef, fullUrl);
-  } else {
-    res.sendStatus(StatusCodes.NOT_FOUND); // 404
-  }
-});
+router.post('/:fullUrl', notificationService.reportTrigger);
 
 /**
  * Notification from EHR of changed data to start workflow process. The
  * id param is the PlanDefinition if which defines the workflow. The body
  * contains the triggering resource.
  */
-router.put('/:fullUrl/:resource/:resourceId', (req, res) => {
-  const planDef = getPlanDef(req.params.fullUrl);
-  if (planDef) {
-    res.sendStatus(StatusCodes.OK);
-    const serverUrl = getBaseUrlFromFullUrl(base64url.decode(req.params.fullUrl));
-    const fullUrl = `${serverUrl}/${req.params.resource}/${req.params.resourceId}`;
-    debug(`Received ${req.params.resource}/${req.params.resourceId} from subscription ${fullUrl}`);
-    const collection = `${req.body.resourceType.toLowerCase()}s`;
-    db.upsert(collection, { fullUrl, ...req.body }, r => r.fullUrl === fullUrl);
-    startReportingWorkflow(planDef, serverUrl, req.body);
-  } else {
-    res.sendStatus(StatusCodes.NOT_FOUND); // 404
-  }
-});
-
-/**
- * Retrieves the PlanDefinition with the given id from the db
- *
- * @param {string} fullUrl - the base 64 url encoded fullUrl of the PlanDefinition
- * @returns the PlanDefinition with the given id or null if not found
- */
-function getPlanDef(fullUrl) {
-  const decodedFullUrl = base64url.decode(fullUrl);
-  const resultList = db.select(PLANDEFINITIONS, s => s.fullUrl === decodedFullUrl);
-  if (resultList[0]) return resultList[0];
-  else return null;
-}
+router.put('/:fullUrl/:resource/:resourceId', notificationService.reportTrigger);
```

Comment:
Same as the other endpoint, I don't see `resourceId` being used in `reportTrigger`